src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java

```diff
@@ -83,9 +105,105 @@ public String getScript() {
 
     @SuppressFBWarnings(value="VA_FORMAT_STRING_USES_NEWLINE", justification="%n from master might be \\n")
     @Override protected FileMonitoringController doLaunch(FilePath ws, Launcher launcher, TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
-        List<String> args = new ArrayList<String>();
+
+        FilePath nodeRoot = getNodeRoot(ws);
+        AgentInfo agentInfo = getAgentInfo(nodeRoot);
         PowershellController c = new PowershellController(ws);
 
+        List<String> powershellArgs = new ArrayList<>();
+        if (!loadProfile) {
+            powershellArgs.add("-NoProfile");
+        }
+        powershellArgs.add("-NonInteractive");
+        if (!launcher.isUnix()) {
+            powershellArgs.addAll(Arrays.asList("-ExecutionPolicy", "Bypass"));
+        }
+
+        if (launcher.isUnix() || "pwsh".equals(powershellBinary)) {
+            usesBom = false;
+        }
+
+        List<String> launcherCmd = null;
+        FilePath binary;
+        // Binary does not support pwsh on linux
+        boolean pwshLinux;
+        if ((agentInfo.getOs() == AgentInfo.OsType.LINUX) && "pwsh".equals(powershellBinary)) {
+            pwshLinux = true;
+        } else {
+            pwshLinux = false;
+        }
+        if (FORCE_BINARY_WRAPPER && !pwshLinux && (binary = requestBinary(nodeRoot, agentInfo, ws, c)) != null) {
```

Comment:
NIT: its not "FORCING" if it is conditional on something else...
"FORCE" is I don't care about the environment or any other factors I am doing this regardless.

Reply:
Good point. I'll go with `USE_BINARY_WRAPPER` or `USE_SCRIPT_WRAPPER` depending if we make the binary the default